libtsuba/include/tsuba/RDG.h

```diff
@@ -63,41 +68,100 @@ class KATANA_EXPORT RDG {
   /// @param handle :: handle indicating where to store RDG
   /// @param command_line :: added to metadata to track lineage of RDG
   /// @param versioning_action :: can be set to 'RDG::RDGVersioningPolicy::IncrementVersion' or
-  /// @param ff :: if ff is not nullptr, it is persisted as the topology for this RDG.
-  /// 'RDG::RDGVersioningPolicy::RetainVersion' to indicate whether RDG version is
+  /// 'RDG::RDGVersioningPolicy::RetainVersion' to indicate whether RDG version is changing with this store.
+  /// @param topology_ff :: if not nullptr, it is persisted as the topology for this RDG.
+  /// @param node_entity_type_id_array_ff :: if not nullptr, it is persisted as the node_entity_type_id_array for this RDG.
+  /// @param edge_entity_type_id_array_ff :: if not nullptr, it is persisted as the edge_entity_type_id_array for this RDG.
+  /// @param node_entity_type_id_dictionary :: persisted as the node EntityTypeID -> Atomic node EntityType id mapping
+  /// @param edge_entity_type_id_dictionary :: persisted as the edge EntityTypeID -> Atomic edge EntityType id mapping
+  /// @param node_entity_type_id_name :: persisted as the node EntityTypeID -> node EntityTypeID name mapping
+  /// @param edge_entity_type_id_name :: persisted as the edge EntityTypeID -> edge EntityTypeID name mapping
   katana::Result<void> Store(
       RDGHandle handle, const std::string& command_line,
-      RDGVersioningPolicy versioning_action, std::unique_ptr<FileFrame> ff);
+      RDGVersioningPolicy versioning_action,
+      std::unique_ptr<FileFrame> topology_ff,
+      std::unique_ptr<FileFrame> node_entity_type_id_array_ff,
+      std::unique_ptr<FileFrame> edge_entity_type_id_array_ff,
+      const std::unordered_map<EntityTypeID, std::vector<EntityTypeID>>&
```

Comment:
I'm going to push back against maps and vectors as the interface between PropertyGraph and RDG. I like that arrays are being passed as FileFrames. That's good. And that we don't create a FileFrame from NUMAArray directly. The idea being that data structure choices are contained. 

Once we expose the data structure choices, a decision to change internal data structure is no longer contained to the class or even layer (logically, layers being in-memory representation and for-storage representation). 

In short, can we pass maps as something generic. For example iterator range over pairs. You can construct a map from those pairs at either end.

Reply:
the libtsuba side of this requested change has been made